### DISCUSSION TITLE
Fluent UI Icon: Render an alternate icon in case the icon name provided is wrong

### DISCUSSION THREAD

**Discussion by utkarsh2019-msft** [microsoft/fluentui#24335] [Q&A] [answered]
Hello @fabricteam , I am trying to find out if there is any capability in Fluent UI Icon component to render an alternative icon or a default icon in case the icon name provided is not an existing icon? I am asking this because I am currently working on a feature where the icon name is being set by an API value and I want to set a default icon name in case the API returns icon names that don't exist.

Thanks for all your help.

**Comment by micahgodbolt** [comment, marked as answer]
There is nothing built into the component to allow for an alternative icon to be used. You might consider wrapping the Icon in a component of your own that checks for the icon availability before rendering the Fluent UI icon.
